release/benchmark-worker-startup/benchmark_worker_startup.py

```diff
@@ -0,0 +1,363 @@
+                        "perf_metric_type": "LATENCY",
+                    }
+                )
+
+            results[test_name] = test_summary
+
+        results["perf_metrics"] = perf_metrics
+        return results
+
+
+def print_disk_config():
+    print("Getting disk sizes via df -h")
+    subprocess.check_call("df -h", shell=True)
+
+def measure_disk_speed():
+    subprocess.check_call("sudo hdparm -t /dev/nvme0n1", shell=True)
+
+def generate_test_matrix(
+    num_cpus_in_cluster: int,
+    num_gpus_in_cluster: int,
+    num_tasks_or_actors_per_run: int,
+    num_measurements_per_test: int,
+):
+
+    num_repeated_jobs_or_runs = num_measurements_per_test
+    total_num_tasks_or_actors = num_tasks_or_actors_per_run * num_repeated_jobs_or_runs
+
+    num_jobs_per_type = {
+        "cold_start": num_repeated_jobs_or_runs,
+        "warm_start": 1,
+    }
+
+    imports_to_try = ["torch", "none"]
+
+    tests = set()
+
+    for with_tasks in [True, False]:
+        for with_gpu in [True, False]:
+            for with_runtime_env in [True, False]:
+                for import_to_try in imports_to_try:
+                    for num_jobs in num_jobs_per_type.values():
+
+                        num_tasks_or_actors_per_job = (
+                            total_num_tasks_or_actors // num_jobs
+                        )
+                        num_runs_per_job = (
+                            num_tasks_or_actors_per_job // num_tasks_or_actors_per_run
+                        )
+
+                        test = TestConfiguration(
+                            num_jobs=num_jobs,
+                            num_runs_per_job=num_runs_per_job,
+                            num_tasks_or_actors_per_run=num_tasks_or_actors_per_run,
+                            with_tasks=with_tasks,
+                            with_gpu=with_gpu,
+                            with_runtime_env=with_runtime_env,
+                            import_to_try=import_to_try,
+                            num_cpus_in_cluster=num_cpus_in_cluster,
+                            num_gpus_in_cluster=num_gpus_in_cluster,
+                            num_nodes_in_cluster=1,
+                        )
+                        tests.add(test)
+
+    return tests
+
+
+@dataclass(eq=True, frozen=True)
+class TestConfiguration:
+    num_jobs: int
+    num_runs_per_job: int
+    num_tasks_or_actors_per_run: int
+    with_gpu: bool
+    with_tasks: bool
+    with_runtime_env: bool
+    import_to_try: str
+    num_cpus_in_cluster: int
+    num_gpus_in_cluster: int
+    num_nodes_in_cluster: int
+
+    def __repr__(self):
+        with_gpu_str = "with_gpu" if self.with_gpu else "without_gpu"
+        executable_unit = "tasks" if self.with_tasks else "actors"
+        cold_or_warm_start = "cold" if self.num_jobs > 1 else "warm"
+        with_runtime_env_str = (
+            "with_runtime_env" if self.with_runtime_env else "without_runtime_env"
+        )
+        single_node_or_multi_node = (
```

Comment:
can we revisit this when scaling this benchmark to multiple nodes? It probably should list the configuration passed in via command-line, since `ray.nodes()` can report dead nodes (and filtering by alive nodes can yield incorrect number in rare case where node dies during measurement).